### PR TITLE
Feature/user management

### PR DESCRIPTION
**Pull request by caleb-acosta**
User management API, Repositories and Migrations

### CODE REVIEW

File: Api.Test/UserControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+using Mineshard.Api.Mappings;
+using Mineshard.Api.Models.DTO.User;
+using Mineshard.Api.Models.Reports;
+using Mineshard.Persistence.Models;
+using Mineshard.Persistence.Models.Auth;
+using Mineshard.Persistence.Repos;
+
+using Moq;
+
+using Newtonsoft.Json.Bson;
```
Comment (hochata) [inline Api.Test/UserControllerTest.cs:22]: What is this for?

File: Persistence/RepoAnalysisContext.cs [hunk 2]
```diff
@@ -36,4 +41,32 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
+        var collaboratorRoleId = new Guid("c309fa92-2123-47be-b397-a1c77adb502c");
+
+        var roles = new List<Role>
+            {
+                new Role
+                {
+                    Id = adminRoleId,
+                    Name = "Admin",
+                    Description = "Administrator role",
+                    CreatedAt = DateTime.UtcNow,
```
Comment (hochata) [inline Persistence/RepoAnalysisContext.cs:59]: This should probably be set automatically. 

But for the MR it is okay.

File: Persistence/Repos/SqlUserRepository.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            existingUser.RoleId = user.RoleId;
+            existingUser.Email = user.Email;
+            existingUser.Name = user.Name;
+            _analysisContext.SaveChanges();
+            return existingUser;
+        }
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
```
Comment (hochata) [inline Persistence/Repos/SqlUserRepository.cs:70]: Don't forget to dispose your `_analysisContext`

File: Api.Test/UserControllerTest.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            var id = new Guid("73bc25af-20b1-49bb-ad54-b775b9ec1ae2");
+            User user = UserFixtures.User;
+            this.mockRepo.Setup(u => u.GetById(id)).Returns(user);
+
+            this.mockRepo.Setup(u => u.Update(user)).Callback(() => user.Role = new Role() { Name = "Writer", Description = "Writer", Id = user.RoleId });
+
+            string name = "updated name";
+            string email = "new@mail.com";
+            string username = "updatedUsername";
+            var roleId = Guid.NewGuid();
```
Comment (hochata) [inline Api.Test/UserControllerTest.cs:104]: So the `roleId` isn't validated?

File: Api/Models/DTO/User/UserDto.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+namespace Mineshard.Api.Models.DTO.User
+{
+    public class UserDto
+    {
+        public Guid Id { get; set; }
+        public required string Name { get; set; }
+        public required string Username { get; set; }
+        public required string Email { get; set; }
+
+        public RoleDto? Role { get; set; }
```
Comment (hochata) [inline Api/Models/DTO/User/UserDto.cs:12]: I thought this was just going to be a string.

Any particular reason to have it as another DTO?

File: Api.Test/UserControllerTest.cs (hunk 4 above)
Comment (caleb-acosta) [reply Api.Test/UserControllerTest.cs:104]: But why an enum to hardcode roles?

File: Persistence/RepoAnalysisContext.cs (hunk 2 above)
Comment (caleb-acosta) [reply Persistence/RepoAnalysisContext.cs:59]: What do you mean?

Comment (hochata) [reply Persistence/RepoAnalysisContext.cs:59]: The `CreatedAt` field should probably be a property set on the DB level, not here.

But I don't think that is too relevant now, as there is no way for users to create new roles.

**Pull request merged**